src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -429,6 +432,53 @@ public async Task<ActionResult<Instance>> Delete(int instanceOwnerPartyId, Guid
                     return StatusCode(500, $"Unexpected exception when updating instance after soft delete: {e.Message}");
                 }
             }
+        }        
+
+        /// <summary>
+        /// Add to an instance that a given stakeholder considers the instance as no longer needed by them. The stakeholder has
+        /// collected all the data and information they needed from the instance and expect no additional data to be added to it.
+        /// </summary>
+        /// <param name="instanceOwnerPartyId">The party id of the instance owner.</param>
+        /// <param name="instanceGuid">The id of the instance whos process history to retrieve.</param>
+        /// <returns>Returns a list of the process events.</returns>        
+        [Authorize(Policy = AuthzConstants.POLICY_INSTANCE_COMPLETE)]
+        [HttpPost("{instanceOwnerPartyId:int}/{instanceGuid:guid}/complete")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [Produces("application/json")]
+        public async Task<ActionResult<Instance>> AddCompleteConfirmation(
+            [FromRoute] int instanceOwnerPartyId,
+            [FromRoute] Guid instanceGuid)
+        {
+            string instanceId = $"{instanceOwnerPartyId}/{instanceGuid}";
+
+            Instance instance = await _instanceRepository.GetOne(instanceId, instanceOwnerPartyId);
+
+            string org = User.GetOrg();
+
+            instance.CompleteConfirmations ??= new List<CompleteConfirmation>();
+            if (instance.CompleteConfirmations.Any(cc => cc.StakeholderId == org))
+            {
+                instance.SetPlatformSelflink(_storageBaseAndHost);
+                return Ok(instance);
+            }
+
+            instance.CompleteConfirmations.Add(new CompleteConfirmation { StakeholderId = org, ConfirmedOn = DateTime.UtcNow });
+            instance.LastChanged = DateTime.UtcNow;
+            instance.LastChangedBy = org;
```

Comment:
Last changed by is set as userId or orgNo.. Wouldn't this be setting the three letter code? This would break the implementation in the messagebox amongst other places.

Reply:
Okay. Do we support changes performed by the application owner? How should that be registered? An alternative that I have considered is to not update the LastChanged values at all.

Reply:
We have an extensionMethod that is GetOrgId for the claimsPrincipal I think..  Check one of the other methods in instancesController. They use the same implementation

Reply:
We can get the org number, would that work?

Still not sure if we should update LastChanged at all. I'm not sure if this update should be registered as a change in that way.

Reply:
User.GetOrgNumber() should do the trick here for LastChangedBy